Forge.TreeWalker/contracts/ForgeSchemaDictionaryValidationRules.json

```diff
@@ -0,0 +1,31 @@
+{
+    "type": "object",
+    "patternProperties": {
+        ".*?": {
+            "$ref": "#/definitions/ForgeTreeDefinition"
+        }
+    },
+    "additionalProperties": false,
+    "definitions": {
+        "ForgeTreeDefinition": {
+            "additionalProperties": false,
+            "properties": {
+                "Tree": {
+                    "$ref": "#/definitions/TreeDefinition"
+                },
+                "RootTreeNodeKey": {
+                    "type": "string"
+                }
+            },
+            "type": "object"
+        },
+        "TreeDefinition": {
+            "type": "object",
+            "patternProperties": {
+                ".*?": {
+                    "$dynamicRef": "file://ForgeSchemaValidationRules.json#/definitions/TreeDefinition"
```

Comment:
>$dynamicRef [](http://example.com/codeflow?start=21&length=11)

Can we use $ref here instead? I'm less familiar with dynamicRef, but some quick research is saying it is an advanced feature and to use it with caution. Sounds scary if we can avoid it. I don't see why $ref wouldn't just work, since it's working in our app's rules today.